woodwork/column_accessor.py

```diff
@@ -217,10 +219,15 @@ def wrapper(*args, **kwargs):
     def _validate_logical_type(self, logical_type):
         """Validates that a logical type is consistent with the series dtype. Performs additional type
         specific validation, as required."""
-        if logical_type.pandas_dtype != str(self._series.dtype):
+        valid_dtype = _get_valid_dtype(self._series, logical_type)
+        if valid_dtype != str(self._series.dtype):
+            if ks and isinstance(self._series, ks.Series) and logical_type.backup_dtype:
+                convert_dtype = logical_type.backup_dtype
+            else:
+                convert_dtype = logical_type.pandas_dtype
```

Comment:
This is needed because of a Koalas quirk. If you run `astype('str')` on a koalas series you will get back a Series with `object` dtype. However, if you run `.astype('object')` on a Koalas series you will get an error. 

In the case that we are using Koalas and `valid_dtype` ends up being `object` we want to tell the user to try to run `astype('str')`, corresponding to the `backup_dtype` defined on the LogcialType, so we can't just use the value of `valid_dtype` in the error message.

Reply:
Okay, so if I'm understanding: We define some backup dtypes as `str` but the way the dtype is stored is as `object`, so when checking the if the underlying dtype would have to change to init the Woodwork typing info, we need to look for `object`, but the actual conversion is still to `str` so that's what we want to tell users about?

I see we have a comment in `_get_valid_dtype` about this. Can we add one here as well?

Reply:
Yes, you have understood correctly. I can add a comment here as well.